src/js/components/tabs.js

```diff
@@ -0,0 +1,220 @@
+
+        break;
+      }
+      case keyCodes.up: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.end: {
+
+        focusList[focusList.length - 1].focus();
+
+        break;
+      }
+      case keyCodes.home: {
+
+        focusList[0].focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabActivated' custom event,
+   * deactivate any other tab, and activate a tab/panel pair. It removes the
+   * panel's hidden attribute, adds the .rvt-tab-open class, sets the tab's
+   * aria-selected attribute to true. It also allows developers to setup a
+   * custom callback function.
+   * @param {Function} callback 
+   */
+  activateTab(tab, callback) {
+    // Trigger tabActivated custom event. This event is used to control the process of switching between tabs.
+
+    const activationEvent = dispatchCustomEvent(
+      'tabActivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    this.panels.forEach(panel => {
+      if (panel.getAttribute(this.panelAttribute) !== tab.dataset.tabPanel && !panel.hasAttribute('hidden')) {
+        this.deactivateTab(panel);
+      }
+    });
+
+    // Activate the appropriate tab/panel pair
+    tab.removeAttribute('hidden');
+    tab.classList.add('rvt-tab-open');
+    trigger.setAttribute('aria-selected', 'true');
+    trigger.removeAttribute('tabindex');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabDeactivated' custom event,
+   * and deactivates a tab/panel pair. It adds the hidden attribute, removes
+   * the .rvt-tab-open class, and sets the tab's aria-selected attribute to
+   * false. It also allows developers to setup a custom callback function.
+   * @param {Function} callback 
+   */
+  deactivateTab(tab, callback) {
+    // Trigger tabDeactivated custom event. This event is used to control the process of switching between tabs.
+
+    const deactivationEvent = dispatchCustomEvent(
+      'tabDeactivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!deactivationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    // Deactivate the appropriate tab/panel pair
+    tab.setAttribute('hidden', '');
+    tab.classList.remove('rvt-tab-open');
+    trigger.setAttribute('aria-selected', 'false');
+    trigger.setAttribute('tabindex', '-1');
+
+    if (callback && typeof callback === 'function') {
+      callback();
+    }
+  }
+
+  init() {
+    if (this.openOnInit) { this.activateTab(this.openOnInit); }
```

Comment:
See previous comments on setting `this.openOnInit` during set up in the constructor. With that in mind you won't need this check anymore because `this.openOnInit` will either be the tab that was initialized via a data attribute in the DOM, or the first tab. So this can become:

```js
this.activateTab(this.openOnInit);
```